src/executor/executor.spec.ts

```diff
@@ -93,6 +93,47 @@ describe("Task Executor", () => {
     });
   });
 
+  describe("run()", () => {
+    it("should run all tasks even if some fail", async () => {
+      const executor = await TaskExecutor.create({ package: "test", logger, yagnaOptions });
+
+      jest.spyOn(Task.prototype, "isFinished").mockImplementation(() => true);
+      const onErrorSpy = jest.fn();
+      const executorEndSpy = jest.spyOn(executor as any, "doEnd");
+
+      const rejectedSpy = jest.spyOn(Task.prototype, "isRejected");
+      const resultsSpy = jest.spyOn(Task.prototype, "getResults");
+      const errorSpy = jest.spyOn(Task.prototype, "getError");
+
+      rejectedSpy.mockImplementationOnce(() => false);
+      resultsSpy.mockImplementationOnce(() => "result 1");
+      const result1 = await executor.run(() => Promise.resolve());
+
+      try {
+        rejectedSpy.mockImplementationOnce(() => true);
+        errorSpy.mockImplementationOnce(() => new Error("test"));
+        await executor.run(() => Promise.resolve());
```

Comment:
This can be replaced with something like `await expect(executor.run(() => throw new Error("test")).rejects.toEqual("test")`, then you can get rid of the whole `try/catch` block here. Then the `onError` spy also becomes unnecessary.

Reply:
Good idea. I also changed the other calls to `executor.run` to use `await expect(...)` for consistency